diff --git a/fsck.te b/fsck.te
index d5a6db1..c54ab7c 100644
--- a/fsck.te
+++ b/fsck.te
@@ -42,6 +42,6 @@ neverallow fsck {
 }:blk_file no_rw_file_perms;
 
 # Only allow entry from init or vold via fsck binaries
-neverallow { domain -init -vold } fsck:process transition;
-neverallow * fsck:process dyntransition;
+neverallow { domain -init -vold -su } fsck:process transition;
+neverallow { domain -su } fsck:process dyntransition;
 neverallow fsck { file_type fs_type -fsck_exec }:file entrypoint;
