diff --git a/vold.te b/vold.te
index fb3673c..b281442 100644
--- a/vold.te
+++ b/vold.te
@@ -188,10 +188,10 @@ allow vold toolbox_exec:file rx_file_perms;
 # Prepare profile dir for users.
 allow vold user_profile_data_file:dir create_dir_perms;
 
-neverallow { domain -vold } vold_data_file:dir ~{ open create read getattr setattr search relabelto ioctl };
-neverallow { domain -vold } vold_data_file:notdevfile_class_set ~{ relabelto getattr };
-neverallow { domain -vold -init } vold_data_file:dir *;
-neverallow { domain -vold -init } vold_data_file:notdevfile_class_set *;
-neverallow { domain -vold -init } restorecon_prop:property_service set;
+neverallow { domain -vold -su } vold_data_file:dir ~{ open create read getattr setattr search relabelto ioctl };
+neverallow { domain -vold -su } vold_data_file:notdevfile_class_set ~{ relabelto getattr };
+neverallow { domain -vold -init -su } vold_data_file:dir *;
+neverallow { domain -vold -init -su } vold_data_file:notdevfile_class_set *;
+neverallow { domain -vold -init -su } restorecon_prop:property_service set;
 
 neverallow vold fsck_exec:file execute_no_trans;
